snap/snapcraft.yaml

```diff
@@ -50,17 +59,26 @@ apps:
 
   # GTK3+ example
   gtk3-test:
-    command: bin/wrapper mir_demo_server --test-client $SNAP/gtkapp.sh
+    command-chain:
+      - bin/desktop-launch
+      - bin/wrapper
+    command: usr/bin/mir_demo_server --test-client $SNAP/gtkapp.sh
     environment:
       PATH: $SNAP/bin/:$SNAP/usr/bin/:${SNAP}/usr/games:${PATH}
 
   performance-test:
-    command: bin/performance-env.sh wrapper mir_performance_tests
+    command-chain:
+      - bin/performance-env.sh
+      - bin/desktop-launch
+      - bin/wrapper
+    command: usr/bin/mir_performance_tests
     plugs:
       - network-bind
 
   # For experimenting
   bash:
+    command-chain:
+      - bin/desktop-launch
     command: bin/wrapper bash
```

Comment:
Not this?

```suggestion
    command-chain:
      - bin/desktop-launch
      - bin/wrapper
    command: bash
```

Reply:
```plain
Failed to generate snap metadata: The specified command 'bash' defined in the app 'bash' is not executable.
```

Reply:
TBH you should just get rid of that now. `snap run --shell mir-test-tools.*` becomes just that, thanks to `command-chain`.

Reply:
```diff
diff --git a/wrapper/wrapper b/wrapper/wrapper
index 059f919..f8582ca 100755
--- a/wrapper/wrapper
+++ b/wrapper/wrapper
@@ -18,11 +18,13 @@ else
 fi
 
 if [ "${MIR_SERVER_CONSOLE_PROVIDER}" = "vt" ]; then
-  MIR_SERVER_VT=${MIR_SERVER_VT:-$(snapctl get vt)}
+  export MIR_SERVER_VT=${MIR_SERVER_VT:-$(snapctl get vt)}
 else
   unset MIR_SERVER_VT
 fi
 
 unset WAYLAND_DISPLAY
 
+export MIR_SERVER_CONSOLE_PROVIDER
+
 exec $@
```

This (and maybe similar elsewhere) should do to get rid of the `bash` command after all.

Reply:
Yeah, I built something similar for testing overnight. If that works out on Core and Classic I'll push shortly.

Reply:
> Yeah, I built something similar for testing overnight. If that works out on Core and Classic I'll push shortly.

Well, that much worked, but I discovered another issue: `mir_demo_server` is deleting the link created by `desktop-launch`...
```plain
root@localhost:/home/alan-griffiths# ls -lh $XDG_RUNTIME_DIR/
total 0
lrwxrwxrwx 1 root root 21 Oct 22 08:59 wayland-0 -> /run/user/0/wayland-0
root@localhost:/home/alan-griffiths# $SNAP/usr/bin/mir_demo_server --test-client $SNAP/usr/bin/mir_demo_client_wayland_egl_spinner 
[2020-10-22 09:00:02.171271] <information> mirserver: Starting
...
[2020-10-22 09:00:13.571264] <information> mirserver: Stopping
root@localhost:/home/alan-griffiths# ls -lh $XDG_RUNTIME_DIR/
total 0
```
That's going to mess things up when running tests!

Reply:
> Well, that much worked, but I discovered another issue: `mir_demo_server` is deleting the link created by `desktop-launch`...

Actually, it is overwriting it. (Because `wl_display_add_socket_auto()` doesn't see a corresponding `.lock` file.)